metricbeat/module/kafka/consumergroup/query.go

```diff
@@ -113,12 +114,22 @@ func fetchGroupInfo(
 
 		for topic, partitions := range ret.off.Blocks {
 			for partition, info := range partitions {
+				partitionOffset, err := getPartitionOffsetFromTheLeader(b, topic, partition)
+				if err != nil {
+					logp.Err("failed to fetch offset for (topic, partition): ('%v', %v)", topic, partition)
+					continue
+				}
+				consumerLag := partitionOffset - info.Offset
+				if consumerLag < 0 {
```

Comment:
If this is a valid concern, maybe it's fair to leave a comment notice: why?

Reply:
yeap, you are right! I don't think that we need this, and it also might be misleading to have zeros instead of the true negative value. Removing.